sdk/core/Azure.Core/src/Shared/OperationInternalBase.cs

```diff
@@ -188,20 +189,29 @@ private async ValueTask<Response> WaitForCompletionResponseAsync(bool async, Tim
                 return lockOrValue.Value;
             }
 
-            var poller = new OperationPoller(_fallbackStrategy);
-            var response = async
-                ? await poller.WaitForCompletionResponseAsync(this, pollingInterval, cancellationToken).ConfigureAwait(false)
-                : poller.WaitForCompletionResponse(this, pollingInterval, cancellationToken);
+            using var scope = CreateScope(string.IsNullOrEmpty(_operationTypeName) ? string.Empty : $"{_operationTypeName}.WaitForCompletionResponse");
```

Comment:
`_operationTypeName` can be empty only if operation is completed, so result of this check is known upfront. And if it is not empty, we create garbage strings every time scope is created. Considering the time that is required to complete the operation, those strings can end up in Gen2.

Reply:
Makes sense. @annelo-msft  How about we go back to the original implementation, i.e. having two fields `_updateStatusScopeName` and `_waitForCompletionResponseScopeName`? Then it will only have two strings for an operation instance.

Reply:
@pshao25, I believe Alex is saying that we don't need to do the check because we already know the result.  By not doing the check, we would avoid creating strings we don't need that could also end up in Gen2.  @AlexanderSher, is this correct?  Do you have a recommendation for what Pan should to here instead?

Reply:
@pshao25 got it right. We can set values to `_updateStatusScopeName` and `_waitForCompletionResponseScopeName` in constructor or when they are accessed. There is a case when operation is created and not awaited, but that case is rare. The most common case is when both fields are used, so I think it is better to assign these fields in constructor. In this case, `_operationTypeName` won't be required.

Reply:
Thank you for the clarification, @AlexanderSher.  I am fine with your recommended implementation and we can come back and change it if needed when we refactor the LRO types in MQ.

Reply:
Do we have further plan to refactor? Could you let me know the context?

Reply:
- https://github.com/Azure/autorest.csharp/issues/1706
- https://github.com/Azure/azure-sdk-for-net/issues/25991
- https://github.com/Azure/autorest.csharp/issues/2231